### PR TITLE
Fix Ensemble dir creation bug in multi-GPU testing cases

### LINKED ISSUE #6417: Auto3dseg integration
```

predictions
[2023-04-22T04:14:14.404Z] 2023-04-22 04:14:13,627 - INFO - Directory /tmp/tmp8m_dx6rq/work_dir/ensemble_output is created to save ensemble predictions
[2023-04-22T04:14:14.404Z] Traceback (most recent call last):
[2023-04-22T04:14:14.404Z]   File "/opt/conda/lib/python3.8/runpy.py", line 194, in _run_module_as_main
[2023-04-22T04:14:14.404Z]     return _run_code(code, main_globals, None,
[2023-04-22T04:14:14.404Z]   File "/opt/conda/lib/python3.8/runpy.py", line 87, in _run_code
[2023-04-22T04:14:14.404Z]     exec(code, run_globals)
[2023-04-22T04:14:14.404Z]   File "/home/jenkins/agent/workspace/MONAI-postmerge/monai_github/monai/apps/auto3dseg/__main__.py", line 24, in <module>
[2023-04-22T04:14:14.404Z]     fire.Fire(
[2023-04-22T04:14:14.404Z]   File "/opt/conda/lib/python3.8/site-packages/fire/core.py", line 141, in Fire
[2023-04-22T04:14:14.404Z]     component_trace = _Fire(component, args, parsed_flag_args, context, name)
[2023-04-22T04:14:14.404Z]   File "/opt/conda/lib/python3.8/site-packages/fire/core.py", line 475, in _Fire
[2023-04-22T04:14:14.404Z]     component, remaining_args = _CallAndUpdateTrace(
[2023-04-22T04:14:14.404Z]   File "/opt/conda/lib/python3.8/site-packages/fire/core.py", line 691, in _CallAndUpdateTrace
[2023-04-22T04:14:14.404Z]     component = fn(*varargs, **kwargs)
[2023-04-22T04:14:14.404Z]   File "/home/jenkins/agent/workspace/MONAI-postmerge/monai_github/monai/apps/auto3dseg/ensemble_builder.py", line 478, in ensemble
[2023-04-22T04:14:14.404Z]     self.set_image_save_transform(**self.kwargs)
[2023-04-22T04:14:14.404Z]   File "/home/jenkins/agent/workspace/MONAI-postmerge/monai_github/monai/apps/auto3dseg/ensemble_builder.py", line 438, in set_image_save_transform
[2023-04-22T04:14:14.404Z]     os.makedirs(output_dir)
[2023-04-22T04:14:14.404Z]   File "/opt/conda/lib/python3.8/os.py", line 223, in makedirs
[2023-04-22T04:14:14.404Z]     mkdir(name, mode)
[2023-04-22T04:14:14.404Z] FileExistsError: [Errno 17] File exists: '/tmp/tmp8m_dx6rq/work_dir/ensemble_output'
[2023-04-22T04:14:14.404Z] 2023-04-22 04:14:13,682 - INFO - Auto3Dseg picked the following networks to ensemble:
[2023-04-22T04:14:14.404Z] 2023-04-22 04:14:13,682 - INFO - segresnet2d_0
[2023-04-22T04:14:14.404Z] 2023-04-22 04:14:13,682 - INFO - Auto3Dseg ensemble prediction outputs will be saved in /tmp/tmp8m_dx6rq/work_dir/ensemble_output.
[2023-04-22T04:14:14.404Z] 
Ensembling (rank 0)...:   0%|          | 0/1 [00:00<?, ?it/s]monai.transforms.io.dictionary LoadImaged.__init__:image_only: Current default value of argument `image_only=False` has been deprecated since version 1.1. It will be changed to `image_only=True` in version 1.3.
[2023-04-22T04:14:14.659Z] [info] checkpoint /tmp/tmp8m_dx6rq/work_dir/segresnet2d_0/model_fold0/best_metric_model.pt loaded
[2023-04-22T04:14:15.589Z] 2023-04-22 04:14:15,372 INFO image_writer.py:197 - writing: /tmp/tmp8m_dx6rq/work_dir/segresnet2d_0/prediction_testing/val_001.fake/val_001.fake_seg.nii.gz

```

### PR DESCRIPTION
**Pull request by mingxin-zheng**
Fixes #6417 .

### Description

- Move the logics of creating ensemble output dir from `ensemble` to `__init__` in EnsembleRunner to avoid processes competing with each other in dir creation
- Add checks in AutoRunner `set_image_save_transform`

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/utils/misc.py [hunk 1]
```diff
@@ -800,3 +801,23 @@ def _calculate_conversion_factor(self):
...
+
+    Args:
+        cls: the class to check.
+        kwargs: kwargs to examine.
+
+    Returns:
+        a boolean inidicating if all keys exist.
+        a set of extra keys that are not used in the __init__.
+    """
+    init_signature = inspect.signature(cls.__init__)
```
Comment (wyli) [inline monai/utils/misc.py:818]: Perhaps this test can be simplified using https://docs.python.org/3/library/inspect.html#inspect.Signature.bind

**Review by wyli — APPROVED** [review]

File: monai/utils/misc.py (hunk 1 above)
Comment (mingxin-zheng) [reply monai/utils/misc.py:818]: I am not confident using the `bind` and `partial_bind` there. And I have a feeling that some try-except may be needed to find the "extra_args". Please feel free to commit to this branch.

**Pull request merged**